Description: Fix the prototype of SplFileInfo::fscanf so that it can be
 extended.
Origin: http://svn.php.net/viewvc?view=revision&revision=298475
Last-Update: 2010-05-02

Index: php/ext/spl/spl_directory.c
===================================================================
--- php.orig/ext/spl/spl_directory.c
+++ php/ext/spl/spl_directory.c
@@ -2570,9 +2570,8 @@ ZEND_BEGIN_ARG_INFO_EX(arginfo_file_obje
 	ZEND_ARG_INFO(0, allowable_tags)
 ZEND_END_ARG_INFO()
 
-ZEND_BEGIN_ARG_INFO_EX(arginfo_file_object_fscanf, 1, 0, 1) 
+ZEND_BEGIN_ARG_INFO_EX(arginfo_file_object_fscanf, 0, 0, 1) 
 	ZEND_ARG_INFO(0, format)
-	ZEND_ARG_INFO(1, ...)
 ZEND_END_ARG_INFO()
 
 ZEND_BEGIN_ARG_INFO_EX(arginfo_file_object_fwrite, 0, 0, 1) 
Index: php/ext/spl/tests/bug51532.phpt
===================================================================
--- /dev/null
+++ php/ext/spl/tests/bug51532.phpt
@@ -0,0 +1,14 @@
+--TEST--
+SPL: Allow valid extension of SplFileObject::fscanf
+--FILE--
+<?php
+
+class A extends SplFileObject {
+    public function fscanf($format) {
+
+    }
+}
+?>
+===DONE===
+--EXPECT--
+===DONE===
